tests/drivers/w1/w1_api/boards/max32655evkit_max32655_m4.overlay

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2023-2024 Analog Devices, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&owm_io_p0_6 {
+	power-source=<MAX32_VSEL_VDDIOH>;
+};
+
+&owm_pe_p0_7 {
+	power-source=<MAX32_VSEL_VDDIOH>;
+};
+
+w1_0: &w1 {
+	status = "okay";
+	pinctrl-0 = <&owm_io_p0_6 &owm_pe_p0_7>;
+	pinctrl-names = "default";
+	internal-pullup = <1>;
+	external-pullup = <0>;
+	long-line-mode = <0>;
+};
```

Comment:
It is because of number of limited 1W GPIOs (2 pins)  and multiple available alternate functions (see below table for a device) on EVKITs, 
So it might not be good option to enforce a default 1W pins for EVKITs... Anyway both option has pros and cons...
![image](https://github.com/user-attachments/assets/0b5e059e-d13b-420a-abcf-3aa3536c62fd)

Moved and removed power selection which are not necessary for the w1 test.